pytorch_lightning/trainer/trainer.py

```diff
@@ -1661,19 +1661,28 @@ def _on_train_batch_end(self, outputs: STEP_OUTPUT, batch, batch_idx, dataloader
             else:
                 callback.on_train_batch_end(self, self.lightning_module, outputs, batch, batch_idx)
 
-    def _call_callbacks_on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[str, dict]:
-        """Called when saving a model checkpoint.
+    def _call_callbacks_state_dict(self) -> Dict[str, dict]:
+        """Called when saving a model checkpoint, calls and returns every callback's `state_dict`, keyed by
+        `Callback.state_key`."""
+        callback_state_dicts = {}
+        for callback in self.callbacks:
+            state_dict = callback.state_dict()
+            if state_dict:
+                callback_state_dicts[callback.state_key] = state_dict
+        return callback_state_dicts
+
+    def _call_callbacks_on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        """Called when saving a model checkpoint, calls every callback's `on_save_checkpoint` hook.
 
-        Calls every callback's `on_save_checkpoint` hook. We have a dedicated function for this rather than using
-        `_call_callback_hooks` because we have special logic for returning callback_states.
+        Will be removed in v1.8: If state is returned, we insert the callback state into
```

Comment:
will this whole method be removed in v1.8?

Reply:
not the whole method, just the part where we overwrite state:
https://github.com/PyTorchLightning/pytorch-lightning/blob/b5b951b05af8811f3e78636f17285f60a2a22e4f/pytorch_lightning/trainer/trainer.py#L1689-L1695